### ISSUE TITLE
Add support for watching files on OSX and Linux

### ISSUE THREAD

**Issue by dfawcus** [Dyalog/link#23]
I note that the Readme mentions that having automatic updates to the workspace based upon filesystem changes and additions only works under windows.

Note that both xnu (the darwin/osx kernel) and the Linux kernel provide for userspace notifications upon changes to specific files/directories.  So it should be possible to implement equivalent functionality on those platforms.

However, maybe this requires changes in the backend APL interpreter process, rather than in this codebase?

**Comment by abrudz** [comment]
Thank you for your interest. Yes, it is of course possible to implement on macOS and Linux, and is indeed planned. The Link system is still under heavy development.

**Comment by mkromberg** [comment]
Support for a file system watcher is currently scheduled for Dyalog version 18.0.

**Comment by rikedyp** [comment]
Current status:
``` bash
mkdir tmp
dyalog
```

```APL
      ]create # tmp
Linked: # ←→ /home/rikedyp/tmp; 1 imports failed (⎕SE.Link.Links[1]
      .inFail)
```

In text editor
```
Dfn←{':'⍵}
```
Save as `tmp/Dfn.aplf`. The function is automatically fixed in the workspace.
```APL
      Dfn 3
: 3
```

No further modifications to `tmp/Dfn.aplf` (in the text editor) are reflected in the workspace. However, modifications using `)ED` the editor *are* reflected in the text file

  ]version
 Dyalog  18.0.38364 64-bit Unicode, BuildID 4c94bea1
 OS      Linux 5.3.0-51-generic #44~18.04.2-Ubuntu SMP Thu Apr 23 1
 SALT    2.807
 UCMD    2.4
 .NET    3.1.3
 WS      18.0

**Comment by rikedyp** [comment]
As for tradfns
```
r ← Tradfn arg
r ← 'r:' arg
```
Causes the following message
```APL
Unable to fix file /home/rikedyp/tmp/Tradfn.aplf
       EM       VALUE ERROR
       Message  No result was provided when the context expected one
```
But the function is fixed and ready to use
```APL
      Tradfn 3
 r:  3
```
Once again, further changes made in the text editor are **not** reflected in the active workspace.

**Comment by abrudz** [comment]
Fixed in 18.0 (uses .NET Core)
